File: src/main/java/com/matag/cards/CardsConfiguration.java
package com.matag.cards;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;

import java.io.File;

import static com.fasterxml.jackson.module.kotlin.ExtensionsKt.jacksonObjectMapper;

@Configuration
@ComponentScan(basePackageClasses = {CardsConfiguration.class})
public class CardsConfiguration {
  public static String getResourcesPath() {
    return new File("src/main/resources").getAbsolutePath();
  }

  @Bean
  public ObjectMapper cardsObjectMapper() {
    return jacksonObjectMapper();
  }
}

File: src/main/kotlin/com/matag/cards/CardsConfiguration.kt
package com.matag.cards

import com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import java.io.File

@Configuration
@ComponentScan(basePackageClasses = [CardsConfiguration::class])
open class CardsConfiguration {
    @Bean
    open fun cardsObjectMapper(): ObjectMapper {
        return jacksonObjectMapper().configure(FAIL_ON_UNKNOWN_PROPERTIES, false)
    }

    companion object {
        fun getResourcesPath(): String = File("src/main/resources").absolutePath
    }
}
